### ISSUE TITLE
Code listing does not format correctly

### ISSUE THREAD

**Issue by GuillaumeDehaene** [quarto-dev/quarto-cli#7253]
### Bug description

Hello everyone,

When rendering to PDF, the code listing float rarely gets rendered incorrectly.

- normally, a numbered code listing has the following structure:
  - Horizontal line
  - Listing $NUM $TITLE
  - second horizontal line
  - $CONTENT
  - final horizontal line
  - grey line on the side
- In bugged listings:
  - content overflows beneath the final horizontal line
  - additional normal text gets written beneath the overflowing content
  - grey line on the side is missing

The bug triggers somewhat randomly when the float gets handled incorrectly by the latex engine.
I have uploaded a small example: https://github.com/GuillaumeDehaene/quarto_bug_report_listings
Reproduce with: `quarto render example.qmd`

Here are some observations about the bug:

- the bug is probably linked to how Latex processes the float
- the bugged box is missing the grey line on the side
- manipulating the text changes the processing of the float and often corrects the bug
- the content of the float doesn't appear to affect the bug (unless the float gets placed in a different position)

After additional investigation:
- it's possible that Latex just incorrectly computes the limit of the float. In my example, the text below the float overlaps the last line of the code listing.
- editing the .tex to remove the `Shaded` environment from the float removes the bug

After additional investigation:
- the Shaded environment is a standin for tcolorbox:
  ```
  \ifdefined\Shaded\renewenvironment{Shaded}{\begin{tcolorbox}[borderline west={3pt}{0pt}{shadecolor}, interior hidden, enhanced, frame hidden, breakable, sharp corners, boxrule=0pt]}{\end{tcolorbox}}\fi
  ```
- removing the `breakable` option fixes the issue
- my overall guess is that the bug occurs when latex decides to break the environment between two pages
- my opinion should be taken with a massive grain of salt: I really struggle with latex

### Steps to reproduce

I have uploaded a small example: https://github.com/GuillaumeDehaene/quarto_bug_report_listings
Reproduce with: `quarto render example.qmd`

`example.pdf` and `example.tex` are generated as outputs of `quarto render example.qmd`

As discussed earlier, the tex file can be manually patched. The repository includes a fixed version of `example.tex` which is named `patched.tex`. `patched.tex` was generated by xelatex.

### Expected behavior

normally, a numbered code listing has the following structure:
  - Horizontal line
  - Listing $NUM $TITLE
  - second horizontal line
  - $CONTENT
  - final horizontal line

### Actual behavior

In bugged listings:
  - content overflows beneath the final horizontal line
  - additional normal text gets written beneath the overflowing content
  - grey line on the side is missing

### Your environment

- IDE: VSCode
- OS: Windows 11
- Quarto: 1.3.450
- - 

### Quarto check output

[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.3.450
      Path: C:\Users\Guillaume\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.5
      Path: d:/Desktop/Dojo/quarto_math_template/.venv/Scripts/python.exe
      Jupyter: 5.4.0
      Kernels: python3

(-) Checking Jupyter engine render....0.01s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.01s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
[>] Checking Jupyter engine render....OK

[>] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/

**Comment by mcanouil** [comment]
I can't compiled your document using Quarto 1.4.
Also, it's really hard to focus on the issue on a 6 pages document.
Please try to make it small.

**Comment by GuillaumeDehaene** [comment]
I've updated the example to be more minimal.
The example now contains only text and two listings with identical content.
Please let me know if there is anything I can do to make the example easier to reproduce.

In this example, I have checked again that the issue can be patched by removing the `breakable` flag from the `tcolorbox` environment.
An example of the patched tex file is present on my example repo.

I have updated the names of the files of the example repo.
The new names should be transparent.
I'll edit the first post to reflect the name changes.

It's good to know that you don't reproduce the issue when updating to the new version.
However, given how random this bug is, it is also possible that it is just being randomly turned off by the latex document being formatted slightly different and the bugged float just being put in a better position.
Is there a recommended way to maintain multiple versions of quarto on the same computer?
If you can please point me to the right way to do this, I'll investigate further to see if I can reproduce the bug with quarto 1.4

Thank you for your help 🙏

**Comment by mcanouil** [comment]
I do not see any issue in the 1.4 pre-release before and now.
<img width="552" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/8242cdd5-1289-41f1-a44d-23f46afbc9e7">

Everything you need is in the documentation.
Pick-up whatever you need, archive for "portable" version, installer, etc.

**Comment by GuillaumeDehaene** [comment]
I've recompiled the document in 1.4. Indeed, the issue is gone, the two listings now are identical in appearance.

This is because the Shaded environment has been redefined to not use the tcolorbox which was causing the issue. It uses snugshade instead.
This changes the appearance of code listings in 1.4 versus 1.3:
- 1.4: the whole background is shaded
- 1.3: only a bar on the side is shaded
In both versions, long code listings overflow the page. apparently the `breakable` flag wasn't achieving much anyway.

My guess would be that the pandoc version change from 1.3 to 1.4 is the reason for the change of behavior and appearance.

Unless the change of appearance requires it, maybe it would be sensible to close this issue since it is fixed in 1.4

**Comment by mcanouil** [comment]
Since you confirmed that the issue is not in 1.4, I am indeed closing it.
Thanks again for the report and the tests!
